app/assets/javascripts/views/courses.js

```diff
@@ -101,5 +111,27 @@ Yacs.views.courses = function (data) {
       c.classList[isSelected ? 'remove' : 'add']('selected');
     });
     if (isCourseSelected(c)) c.classList.add('selected');
+
+    var desc = c.querySelector('course-description');
+    var showButton = c.querySelector('show-hide-button');
+    Yacs.on('click', showButton, function (showButton, event) {
+      var descTruncated = desc.classList.contains("truncated");
+      desc.classList[descTruncated ? 'remove' : 'add']("truncated");
+      showButton.classList[descTruncated ? 'add' : 'remove']("open");
+      event.stopPropagation();
+    });
+    requireTruncation(desc, showButton);
+  });
+
+  var resizingFxn;
+  window.addEventListener("resize", function () {
```

Comment:
One of the main problem with binding the event to window is that the event doesn't expire or reset when we switch to any other view. This is similar to the problem we had before with adding the clear selection button to the header. 

What if we have a function in the header bar, or the main page that manages events like these?